ERCS/erc-7589.md

```diff
@@ -376,9 +402,8 @@ Solidity ABI Specification.
   its arguments (if any) as the identifier. E.g., `keccak256("Player(uint256)")`.
 * The `commitTokens` function MUST revert if the `_tokenAmount` is zero or the `msg.sender` was not approved by the
   `_grantor`. It MAY be implemented as public or external.
-* The `grantRole` function MUST revert if the `_expirationDate` is in the past or if the `msg.sender` is not approved to
- grant roles on behalf of the grantor. It MAY be implemented as public or external.
-* The `revokeRole` function MAY be implemented as public or external and MUST revert if:
+* The `grantRole` function MUST revert if the `_expirationDate` is in the past or if the `msg.sender` is not approved to grant roles on behalf of the grantor. It MAY be implemented as public or external. It is RECOMMENDED using `type(uint64).max` for a permanent role
```

Comment:
The maximum number or characters in a line is 120, otherwise paragraphs will never have an end. Intellij/Webstorm show this by default, vscode might need an extension.

We need to double check all lines changes with this rule